docs/dark.md

```diff
@@ -0,0 +1,163 @@
+
+## DARK
+
+### Main Idea
+
+First, you should read the definitions
+of [diophantine equation](../terms/diophantine_equation.md), [diophantine set](../terms/diophantine_set.md).
+
+The main idea of DARK is to prove a constraint with $\mu$ as the witness, the prover needs to find an auxiliary witness $\omega$, where
+$R_S(\mu, \omega) = 0$ and then uses the integer commitment scheme above to commit to elements of $R_S$.
+
+For example, to prove $\mu
+\geq 0$, the prover
+finds $\omega_{1}, \omega_{2}, \omega_{3}, \omega_{4}$ such that $R_S(\mu,\omega)=\mu - (\omega_1^2 + \omega_2^2 + \omega_3^2 +
+\omega_4^2)$ (see the
+non-negative proof below for more details). Then the prover sends $com(\mu), com(\omega_1),com(\omega_2),com(\omega_3),com(\omega_4)$
+to the verifier. The verifier checks that $R_S(\mu,\omega)=0$ using these commitments.
+
+To put into practical use, there are a few things to consider:
+
+- Compute the witness $\omega$ in polynomial time.
+- The length of our witness has to be sub-quadratic to the length of the input $\mu$
+
+Therefore, we only deal with operations that listed in section [Bounded Arithmetic](#bounded-arithmetic).
+
+### Basic Proofs
+
+> [!NOTE]
+> We can use the [fact](../terms/diophantine_set.md#fact) from Diophantine set to deal with AND/OR operations on prepositions.
+
+Below are the basic proofs for constructing our bounded arithmetic.
+
+#### Non-negative Proof
+
+We will leverage [Lagrange's four-square theorem](https://en.wikipedia.org/wiki/Lagrange%27s_four-square_theorem) to prove that $\mu
+\geq0$. This theorem implies that there exists $\omega_{1},\omega_{2}, \omega_{3}, \omega_{4}\ s.t. \ \mu=\omega_{1}^{2}+\omega_
+{2}^{2}+\omega_{3}^{2}+\omega_{4}^{2}$. So, the representing polynomial is simply $R_S(\mu; \omega_1, \omega_2, \omega_3, \omega_4) =
+\mu - \omega_1^2- \omega_2^2-\omega_3^2- \omega_4^2$.
+
+We can find $\omega_{1},\omega_{2}, \omega_{3}, \omega_{4}$ as follow. If $\mu$ is small, we can brute force over
+$\omega_1,\omega_2,\omega_3\in\lbrack 0, \sqrt{\mu}\rbrack$ and calculate $\omega_4$. If $\mu$ is large, we can use this algorithm:
+
+1. We choose random $\omega_1 \leq \sqrt{\mu},\ \omega_2 \leq \sqrt{\mu−\omega_1^2}$ so that $p \leftarrow \mu−\omega_1^2 −\omega_2^2$
+   is odd.
+2. There exists a good probability that $p$ is a prime number of the form $4k+1$.
+   Using [Fermat's theorem on sums of two squares](https://en.wikipedia.org/wiki/Fermat%27s_theorem_on_sums_of_two_squares), we can
+   find $\omega_{3},\omega_{4}$ efficiently. There are two stages: (i) find $z$ with $z^{2}\equiv -1 \pmod{p}$, and (ii) use $z$ to
+   find $\omega_3$ and $\omega_4$.
+
+   (i). Find $a$ where $a^{(p-1) / 2}\equiv-1\pmod{p}$, then $z=a^{(p-1)/4}$. There are over half of all number $a$ such that $1 < a <
+   p-1$, so taking $a$ at random.
+
+   (ii). Now, we set $\omega_{3}=p \mod z$ and $\omega_{4}=z\mod \omega_3$. If $\omega_{3}^{2}+\omega_{4}^{2} \neq p$, $p$ is not a
```

Comment:
use pmod and \equiv

Reply:
I believe using $mod$ is more appropriate because I want to calculate the value of $\omega$